Shell/ConditionalRedundancyHandler.cpp

```diff
@@ -106,28 +106,17 @@ class ConditionalRedundancyHandler::ConstraintIndex
 
   void insert(const TermStack& ts, void* ptr)
   {
-    static CodeStack code;
-    code.reset();
-
-    static CompileContext cctx;
-    cctx.init();
+    CodeStack code;
+    TermCompiler compiler(code);
     for (const auto& t : ts) {
       if (t.isVar()) {
-        unsigned var=t.var();
-        unsigned* varNumPtr;
-        if (cctx.varMap.getValuePtr(var,varNumPtr)) {
-          *varNumPtr=cctx.nextVarNum++;
-          code.push(CodeOp::getTermOp(ASSIGN_VAR, *varNumPtr));
-        }	else {
-          code.push(CodeOp::getTermOp(CHECK_VAR, *varNumPtr));
-        }
-      }
-      else {
-        ASS(t.isTerm());
-        compileTerm(t.term(), code, cctx, false);
+        compiler.handleVar(t.var());
+        continue;
```

Comment:
Did some of the now missing complexity get hidden elsewhere?

Reply:
Yes, the "code compilation" is now delegated to a `CodeTree::Compiler` instance.

Reply:
Great!